scheds/rust/scx_bpfland/src/bpf/main.bpf.c

```diff
@@ -851,10 +852,45 @@ void BPF_STRUCT_OPS(bpfland_enqueue, struct task_struct *p, u64 enq_flags)
 		kick_idle_cpu(p, tctx, prev_cpu, false);
 }
 
-void BPF_STRUCT_OPS(bpfland_dispatch, s32 cpu, struct task_struct *prev)
+static bool keep_running(const struct task_struct *p, s32 cpu)
 {
-	const struct cpumask *primary = cast_mask(primary_cpumask);
+	const struct cpumask *primary = cast_mask(primary_cpumask), *smt;
+	const struct cpumask *idle_cpumask;
+	struct cpu_ctx *cctx;
+	bool ret;
+
+	/* Do not keep running if the task doesn't need to run */
+	if (!is_queued(p))
+		return false;
+
+	/* Do not keep running if the CPU is not in the primary domain */
+	if (!primary || !bpf_cpumask_test_cpu(cpu, primary))
+		return false;
 
+	/*
+	 * Keep running only if the task is on a full-idle SMT core (or SMT
```

Comment:
I think this gets interesting as the machine becomes more saturated where there are no idle SMTs left. For low saturation this is probably the right thing to do.

Reply:
@hodgesds precisely! I have another patch that I'm still testing to skip searching for idle CPUs when the system is busy :)